src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -1759,7 +1761,31 @@ public String getPackageFromClassName(String className) {
     if (className.contains("<")) {
       className = className.substring(0, className.indexOf("<"));
     }
-    return classAndPackageMap.getOrDefault(className, currentPackage);
+    String pkg = classAndPackageMap.get(className);
+    if (pkg != null) {
+      return pkg;
+    } else {
+      // Check if there is a wildcard import. If there isn't always use
+      // currentPackage.
+      if (wildcardImports.size() == 0) {
+        return currentPackage;
+      }
+      // If there is a wildcard import, check if there is a matching class
+      // in the original codebase in the current package. If so, use that.
+      if (classfileIsInOriginalCodebase(currentPackage + "." + className)) {
+        return currentPackage;
+      }
+      // If not, then check for each wildcard import if the original codebase
+      // contains an appropriate class. If so, use it.
+      for (String wildcardPkg : wildcardImports) {
+        if (classfileIsInOriginalCodebase(wildcardPkg + "." + className)) {
+          return wildcardPkg;
+        }
+      }
+      // If none do, then default to the first wildcard import.
+      String wildcardPkg = wildcardImports.get(0);
```

Comment:
>  I don't think we want Specimin to randomly choose a package to put a synthetic file in

I would not describe this as "random": it deterministically puts the class in the first wildcard import in the file (based on the order of the imports). Since many files will only have one wildcard import, I don't think this is too dangerous.

> Professor, would it be better to throw an exception here? 

I'd be willing to consider throwing an exception if there is more than one wildcard import, but in the case where there is only one it's _almost_ unambiguous, so I'd prefer we handle it cleanly.

That said, I'm okay with Specimin producing a "close but not quite correct" result in non-jar mode, which is what I think this design produces.

Reply:
I see. Should we print out a message or something to inform users? If users get some unexpected results when running the final output of Specimin through a verifier, they can infer where those results originate.

Reply:
I'd be comfortable doing so if we had a proper logging solution. I'll add a TODO in this PR and then we can add logging later.